client/my-sites/domains/domain-management/settings/cards/domain-redirect-card.tsx

```diff
@@ -28,36 +28,40 @@ const noticeOptions = {
 class DomainRedirectCard extends Component {
 	static propTypes = {
 		redirect: PropTypes.object.isRequired,
-		domainName: PropTypes.string.isRequired,
-		targetUrl: PropTypes.string.isRequired,
-		currentRoute: PropTypes.string.isRequired,
+		selectedSite: PropTypes.object.isRequired,
 	};
 
 	state = {
-		redirectUrl: this.props.targetUrl ?? '',
+		targetHost: this.props.redirect.targetHost,
+		secure: this.props.redirect.secure ? 1 : 0,
 	};
 
 	componentDidMount() {
-		this.props.fetchDomainRedirect( this.props.domainName );
+		this.props.fetchDomainRedirect( this.props.selectedSite.domain );
 	}
 
 	componentWillUnmount() {
-		this.props.closeDomainRedirectNotice( this.props.domainName );
+		this.props.closeDomainRedirectNotice( this.props.selectedSite.domain );
 	}
 
 	handleChange = ( event ) => {
-		const redirectUrl = withoutHttp( event.target.value );
-
-		this.setState( { redirectUrl } );
+		const targetHost = withoutHttp( event.target.value );
+		this.setState( { targetHost } );
```

Comment:
I think we're trying to store the whole url in this field (apart from the protocol for some reason).

Reply:
swapped targetHost for targetUrl